CourseTemplate/CourseTemplate.Core/CourseTemplateService.cs

```diff
@@ -21,19 +24,29 @@ public CourseTemplateService()
   
 	public async Task AddAsync(CourseTemplateDto course)
 	{
+		var dbCourseTemplates = await _repository.ListAsync();
+
+		// Check if a course with the same name already exists
+		if (dbCourseTemplates.Any(c => c.CourseName == course.Name))
+		{
+			throw new ArgumentException($"A course with the name '{course.Name}' already exists.");
```

Comment:
Map this error to this code: https://developer.mozilla.org/ru/docs/Web/HTTP/Status/409
You can create custom exception for this